vllm/model_executor/models/gpt_bigcode.py

```diff
@@ -259,10 +265,22 @@ def __init__(
         self.quant_config = quant_config
         self.transformer = GPTBigCodeModel(config, cache_config, quant_config,
                                            lora_config)
-        self.lm_head = self.transformer.wte
         self.unpadded_vocab_size = config.vocab_size
         if lora_config:
             self.unpadded_vocab_size += lora_config.lora_extra_vocab_size
+        self.lm_head = ParallelLMHead(
```

Comment:
This `init` function will call `create_weights()`, which initializes the buffers for the `lm_head`

Are we sure these are properly cleaned up when setting `self.lm_head.weight`?

Additionally, note that `self.lm_head.weight = self.transformer.wte.weight` will only work for fp16 models since there is a more complicated state dict that needs overriding if the lm-head is quantized

I think this is okay (b/c the lm_head was never allowed to be quantized since we do not support quantized lm head for tied embedding models) ... but nonetheless something that could cause errors in the future to be aware of. So perhaps we could enforce that lm head is not quantized?

Reply:
I've now updated the code to assert that no quantization is used. I've also added a subclass of `UnquantizedLinearMethod` that doesn't initialize weights because it's wasteful if the weights are replaced in the next step.